static/src/javascripts/projects/commercial/modules/ipsos-mori.ts

```diff
@@ -24,25 +26,26 @@ const loadIpsosScript = (locale: 'au' | 'uk') => {
  * documentation on DCR: [link](https://github.com/guardian/dotcom-rendering/blob/150fc2d81e6a66d9c3336185e874fc8cd0288546/dotcom-rendering/docs/architecture/3rd%20party%20technical%20review/002-ipsos-mori.md)
  * @returns Promise
  */
-export const init = (): Promise<void> => {
-	return getLocale()
-		.then((locale) => {
-			if (locale === 'GB' || locale === 'AU') {
-				return getInitialConsentState();
-			} else {
-				throw Error('Skipping ipsos process outside GB or AU');
-			}
-		})
-		.then((state) => {
-			if (state.aus) {
-				void loadIpsosScript('au');
-			} else if (getConsentFor('ipsos', state)) {
-				void loadIpsosScript('uk');
-			} else {
-				throw Error('No consent for ipsos in GB');
-			}
-		})
-		.catch((e) => {
-			log('commercial', '⚠️ Failed to execute ipsos', e);
-		});
+export const init = async (): Promise<void> => {
+	const locale = await getLocale();
+	const consentState = await onConsent();
+	const isAU = locale === 'AU' && !!consentState.aus;
+	const isUK = locale === 'GB' && !!consentState.tcfv2;
+
+	if (!isAU && !isUK) {
+		log('commercial', 'Skipping ipsos process outside GB or AU');
+		throw Error('Skipping ipsos process outside GB or AU');
```

Comment:
Just wondering if we need to throw errors here and below rather than just logging.

Do we consider the throws to be failure states or are they just conditions in which we don't load ipsos?

Reply:
Yes, I was wondering whether this was overkill too. I'm thinking that if it's outside the UK and Aus, then it's not a failure state, so the log would be enough. I have more doubt about the other case as the consent state has been checked, and I guess the reason there isn't consent could be failure or that there isn't consent. However, after looking into this one as well, it seems as though there is consent or there isn't. So I guess we could drop both of them.

Reply:
Update: I have removed both